manager/assets/modext/widgets/core/modx.tabs.js

```diff
@@ -1,41 +1,56 @@
 MODx.Tabs = function(config) {
     config = config || {};
-    Ext.applyIf(config,{
-        enableTabScroll: true
-        ,layoutOnTabChange: true
-        ,plain: true
-        ,deferredRender: true
-        ,hideMode: 'offsets'
-        ,defaults: {
-            autoHeight: true
-            ,hideMode: 'offsets'
-            ,border: true
-            ,autoWidth: true
-            ,bodyCssClass: 'tab-panel-wrapper'
-        }
-        ,activeTab: 0
-        ,border: false
-        ,autoScroll: true
-        ,autoHeight: true
-        ,cls: 'modx-tabs'
+    Ext.applyIf(config, {
+        enableTabScroll: true,
+        layoutOnTabChange: true,
+        plain: true,
+        deferredRender: true,
+        hideMode: 'offsets',
+        defaults: {
+            autoHeight: true,
+            hideMode: 'offsets',
+            border: true,
+            autoWidth: true,
+            bodyCssClass: 'tab-panel-wrapper'
+        },
+        activeTab: 0,
+        border: false,
+        autoScroll: true,
+        autoHeight: true,
+        cls: 'modx-tabs'
     });
-    MODx.Tabs.superclass.constructor.call(this,config);
+    MODx.Tabs.superclass.constructor.call(this, config);
     this.config = config;
     this.on({
+        tabchange: function(tabPanel, tab) {
+            /*
+                In certain scenarios, such as when form customization and/or a plugin adds a tab,
+                the state of the Resource tab panel can become uncertain and no tab will be initially
+                selected. This workaround ensures the first tab is selected.
+            */
+            if (this.id === 'modx-resource-tabs' && MODx.request.tab === undefined && !this.getActiveTab()) {
```

Comment:
If we want to try and compose things properly, this sort of edge-case overrides should be in the component that experiences the edge-case. I.e. in the `modx-resource-tabs` component, not here in `MODx.Tabs`. 

However, all of the ExtJS stuff is already messy and not necessarily composed properly in many places, so I'm _very_ low on the SSOGAF about this. If you think this is the easiest way to go about this particular issue, then that's fine and you can mark this resolved without making changes.

Reply:
I'll take a look and let you know later today.

Reply:
Good catch here @Mark-H. Indeed this is better placed in the Resources panel. Done in [this](https://github.com/modxcms/revolution/pull/16369/commits/559f676a2bd24ef938878c033a73907d12ed554c) commit.